### ISSUE TITLE
b-form-input does not update reactively after first input

### ISSUE THREAD

**Issue by maksbotan** [bootstrap-vue/bootstrap-vue#817]
Hi. My problem is as follows. I have this in my template:

```html
<b-form-input type="text" v-model="someField"></b-form-input>
<b-button @click="add">Add<b-button>
```
And a method:
```js
add () {
    this.someField = 'someValue'
}
```
`this.someField` is initially set to `''` in `data`.

What happens: if i click on "Add" before inputing anything to the field, field starts to display "someValue". But if I input anything there and then click the button, field continues to display what I've entered. Vue Devtools inspector shows that `this.someField` is in fact updated to contain `'someValue'`, but `$0.value` in Chrome Devtools shows that actual DOM element still contains the value I've entered, _not_ the one I've set programatically.

If I use raw `<input v-model="someField"/>` in my template, everything works as expected.

Is this a bug in bootstrap-vue or I'm just doing something wrong?

See demonstration in fiddle: https://jsfiddle.net/nz23f2kg/

**Comment by tmorehouse** [comment]
I just updated your fiddle, and it works for me:

https://jsfiddle.net/nz23f2kg/4/

**Comment by mosinve** [comment]
Troy, it dont work for me

**Comment by tmorehouse** [comment]
Weird.  Let me try another browser.

**Comment by alexsasharegan** [comment]
Yeah, the input needs to be reset to the new v-model prop passed down maybe.

**Comment by tmorehouse** [comment]
Yeah, i see that now.  strange.

**Comment by alexsasharegan** [comment]
You can see how it's fine on a native input: https://jsfiddle.net/jv88L704/

**Comment by alexsasharegan** [comment]
I think it would make sense to make use of `v-model` internally.

**Comment by tmorehouse** [comment]
Can't since input (not textarea) has a dynamic `type` attribute (i.e. test, password, number, search, date, etc)

**Comment by mosinve** [comment]
fixed with #821 at dev branch.

**Comment by tmorehouse** [comment]
Fixed in `1.x` branch as well.
